src/charm.py

```diff
@@ -24,22 +28,80 @@ def __init__(self, *args: typing.Any) -> None:
             args: passthrough to CharmBase.
         """
         super().__init__(*args)
+        self._charm_state = CharmState(charm_config=self.config)
+        self._webserver = GunicornWebserver(charm_state=self._charm_state)
         self.framework.observe(self.on.config_changed, self.config_service)
 
+    def flask_container(self, require_connected: bool = True) -> Container:
+        """Get the flask application workload container controller.
+
+        Args:
+            require_connected: if set to ``True``, a runtime exception will be raised if the
+                pebble service inside the container is not ready.
+
+        Return:
+            The controller of the flask application workload container.
+
+        Raises:
+            RuntimeError: if the pebble service inside the container is not ready while the
+                ``require_connected`` is set to True.
+        """
+        container = self.unit.get_container("flask-app")
+        if require_connected and not container.can_connect():
+            raise RuntimeError("pebble inside flask-app container is not ready")
+        return container
+
     def config_service(self, event: ConfigChangedEvent) -> None:
         """Configure the flask pebble service layer.
 
         Args:
             event: the config-changed event that trigger this callback function.
         """
-        container = self.unit.get_container("flask-app")
+        container = self.flask_container(require_connected=False)
         if not container.can_connect():
```

Comment:
Isn't it better to call `flask_container` with require_connected true and catch the exception to defer ?

Reply:
The `RuntimeError` is just a safe guard to avoid accidental invocation without checking. Since the `RuntimeError` can be raised by many other processes, it's a little unsafe to catch that.

Reply:
Well then we can have a custom exception, more specific raised by `flask_container`

Reply:
I created a new method called `flask_container_can_connect` to check if the pebble is ready and removed the `require_connected` argument from `flask_container`.
I think it's too much to create an new entity (custom exception) for this function.